R/reproject_coordinates.R

```diff
@@ -88,52 +86,20 @@ reproject_coordinates <- function(df, col_long, col_lat,
   assert_that(isTRUE(all(map_lgl(df[[col_lat]],  ~ is.numeric(.)))),
               msg = "y coordinates (latitude) should be numbers.")
 
-  if ((all(c(class(crs_input), class(crs_output)) == "CRS")) |
-       all(c(class(crs_input), class(crs_output)) == "crs")) {
-    if (class(crs_input) == "CRS") {
-      if (crs_input@projargs == crs_output@projargs) {
-        warning(paste("Input projection equal to output projection.",
-                      "No reprojection performed."))
-        return(df)
-      }
-    } else {
-      if (crs_input == crs_output) {
-        warning(paste("Input projection equal to output projection.",
-                      "No reprojection performed."))
-        return(df)
-      }
-    }
-  }
-
-  if (class(crs_input) == "CRS") {
-    df_spat <- SpatialPoints(df[c(col_long, col_lat)],
-                             proj4string = crs_input)
-    if (class(crs_output) == "CRS") {
-      df_reproj <- spTransform(df_spat, crs_output)
-    } else {
-      df_spat <- st_as_sf(df_spat, crs = crs_input)
-      df_reproj <- st_transform(df_spat, crs_output)
-    }
-  } else{
-    df_spat <- st_as_sf(df[c(col_long, col_lat)],
-                        coords = c(col_long, col_lat),
-                        crs = crs_input)
-    if (class(crs_output) == "CRS") {
-      df_spat <- as(df_spat, "Spatial")
-      # convert from SpatialPointDataframe to SpatialPoints
-      df_spat <- as(df_spat, "SpatialPoints")
-      colnames(df_spat@coords) <- c(col_long, col_lat)
+  crs_input <- st_crs(crs_input)
+  crs_output <- st_crs(crs_output)
 
-      df_reproj <- spTransform(df_spat, crs_output)
-    } else {
-      df_reproj <- st_transform(df_spat, crs_output)
-    }
+  if (crs_input == crs_output) {
+    warning(paste("Input projection equal to output projection.",
+                  "No reprojection performed."))
```

Comment:
'Reprojection' is actually not an ideal term. That would mean: from one _projected_ CRS to another _projected_ CRS. But general coordinate operations are offered instead, i.e. conversions (within same datum) and transformations (with datum shift). In both cases it can be between geodetic and projected, between geodetic and geodetic, or between projected and projected CRSs.

So better terms would be appropriate (with impact on function names and documentation). You could e.g. say `transform_coordinates()` (like `st_transform()`) to cover both conversions and transformations, since in practice a transformation _pipeline_ is often involved, with conversion ánd transformation steps.

Reply:
See [878d52b](https://github.com/inbo/inborutils/pull/115/commits/878d52b00095c6253ca1b830409d2e10ca0b04f7)